kanidmd/src/lib/credential/mod.rs

```diff
@@ -731,8 +846,8 @@ impl CredentialType {
     fn is_valid(&self) -> bool {
         match self {
             CredentialType::Password(_) | CredentialType::GeneratedPassword(_) => true,
-            CredentialType::PasswordMfa(_, m_totp, webauthn) => {
-                m_totp.is_some() || !webauthn.is_empty()
+            CredentialType::PasswordMfa(_, m_totp, webauthn, _) => {
+                m_totp.is_some() || !webauthn.is_empty() // ignore backup code (it should only be a complement for totp/webauth)
```

Comment:
I think this check won't be needed if you change the code above to only set backup code if totp/webauthn are some. (which they have to be if you are password MFA anyway).

Reply:
Ahhh yes, your right, I misread :)